### PR TITLE
feat(control): add autonomous emergency braking module

### PR DESCRIPTION
**Pull request by purewater0901**
## Description
Currently, autoware does not have any emergency braking systems, and it can be collide obstacles when ego vehicle is out of the lane or ego position is not accurate. In this PR, I created a new package called "autonomous emergency braking", and it enables Autoware to stop ego vehicle when emergency brake needed. 

This new function has the following features.
- Generate predicted path of the ego vehicle from imu data (steering angle) and vehicle velocity data
- Perform collision check with pointcloud data using path generated by the aformentioned method or (and) predicted path from control module
- Send error signals via diagnostics when collision happens

## Tests performed
This function is tested with real vehicle, and the author confirms AEB works correctly.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by TakaHoribe** [comment]
memo: the default is false. https://github.com/autowarefoundation/autoware_launch/pull/232

**Review by TakaHoribe — APPROVED** [review]
LGTM.

Note: this feature could be merged into the node discussed [here](https://github.com/autowarefoundation/autoware.universe/pull/2528) so that some additional features such as the pause interface can be used.

File: system/system_error_monitor/config/system_error_monitor.param.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@
         /autoware/control/autonomous_driving/node_alive_monitoring: default
         /autoware/control/autonomous_driving/performance_monitoring/lane_departure: default
         /autoware/control/control_command_gate/node_alive_monitoring: default
+        /autoware/control/autonomous_emergency_braking/performance_monitoring/emergency_stop: { sf_at: "none", lf_at: "warn", spf_at: "error", auto_recovery: "false"}
```
Comment (ito-san) [inline system/system_error_monitor/config/system_error_monitor.param.yaml:23]: @purewater0901 AEB node seems to be designed to publish OK and ERROR only.
What do you mean by setting `warn` value to `lf_at`?
It is OK that the `warn` is reserved value for future functionality extension to realize a certain use case in the future.

Comment (ito-san) [reply system/system_error_monitor/config/system_error_monitor.param.yaml:23]: I talked to @purewater0901 internally and this `warn` is reserved value to extend functionality of AEB in the future.

**Review by ito-san — APPROVED** [review]
LGTM

**Pull request merged**